tutorials/tour/higher-order-functions.md

```diff
@@ -8,34 +8,112 @@ tutorial: scala-tour
 num: 8
 next-page: nested-functions
 previous-page: anonymous-function-syntax
+assumed-knowledge: sequence-comprehensions
 ---
 
-Scala allows the definition of higher-order functions. These are functions that _take other functions as parameters_, or whose _result is a function_. Here is a function `apply` which takes another function `f` and a value `v` and applies function `f` to `v`:
+Higher order functions take other functions as parameters or return a function as
+a result. This is possible because functions are first-class objects in Scala.
+One of the most common examples is the higher-order
+function `map` which is available for collections in Scala.
+```tut
+val salaries = Seq(20000, 70000, 40000)
+val doubleSalary = (x: Int) => x * 2
+val newSalaries = salaries.map(doubleSalary) // List(40000, 140000, 80000)
+```
+In this case, the function `doubleSalary` gets applied to each element in the
+list of salaries. A more idiomatic way to write the same piece of code would be
 
 ```tut
-def apply(f: Int => String, v: Int) = f(v)
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(_ * 2)
 ```
+The Scala compiler already knows the type of the parameters (a single Int) that function argument
+for `map` needs. Therefore you just need to provide the right side of the function `doubleSalary`. The only
+caveat is that you need to use `_` in place of a parameter name (it was `x` in
+the previous example).
 
-_Note: methods are automatically coerced to functions if the context requires this._
+## Coercing methods into functions
+It is also possible to pass methods as arguments to higher-order functions because
+the Scala compiler will coerce the method into a function.
+```
+case class WeeklyWeatherForecast(temperatures: Seq[Double]) {
 
-Here is another example:
- 
-```tut
-class Decorator(left: String, right: String) {
-  def layout[A](x: A) = left + x.toString() + right
-}
+  private def convertCtoF(temp: Double) = temp * 1.8 + 32
 
-object FunTest extends App {
-  def apply(f: Int => String, v: Int) = f(v)
-  val decorator = new Decorator("[", "]")
-  println(apply(decorator.layout, 7))
+  def getForecastInFahrenheit() = temperatures.map(convertCtoF) // <-- passing the method convertCtoF
```

Comment:
Also, a convention in Scala is to put parens to methods that take no parameter only if this method actually performs side-effects. Since `getForecastInFahrenheit` is pure, you should remove the parens to its definition. Also, I would omit the `get` prefix, which is not common in Scala: `def forecastInFahrenheit: Double`.